packages/middleware/src/apiClientFactory/index.ts

```diff
@@ -102,6 +97,26 @@ const apiClientFactory = <
 
       const isApiFactory = typeof apiOrApiFactory === "function";
       const api = isApiFactory ? apiOrApiFactory(settings) : apiOrApiFactory;
+
+      const namespacedExtensions: Record<string, any> = {};
+      let sharedExtensions = customApi;
+
+      // If the extension is namespaced, we need to merge the extended api methods into the namespace
+      // Otherwise, we can just merge the extended api methods into the api
+      rawExtensions.forEach((extension) => {
+        if (extension.isNamespaced) {
+          namespacedExtensions[extension.name] = {
```

Comment:
Consider this:

```ts
Object.assign(namespacedExtensions[extension.name], extension.extendApiMethods);
```

Makes sense since you're modifying the original object anyway.

Reply:
Is it faster? more readable? optimized somehow? follows best practice?
(I am not triggering you :) I am just asking because there are always multiple ways how a piece of code can be written and I do not see any immediate and obvious benefit from rewriting it.

Reply:
I feel like it's cleaner (I just like one liners)

Reply:
I also think Object.assign is cleaner. Also, it mutates the object, and code with spread creates a new object literal. Anyway I don't know if it's optimized somehow internally and there is significant performance difference